test/L2/L2Reward.t.sol

```diff
@@ -210,141 +219,154 @@ contract L2RewardTest is Test {
+            vm.prank(scenario.stakers[i]);
+            l2Reward.claimRewards(positionsToBeModified);
 
-        checkConsistencyPendingUnlockDailyUnlocked(lockIDs, getLargestExpiryDate(lockIDs));
+            scenario.balances[i] = l2LiskToken.balanceOf(scenario.stakers[i]);
 
-        vm.warp(19740 days + 80 days);
-        // resumes stake 2 on day 80
-        positionsToBeModified[0] = lockIDs[1];
-        vm.prank(stakers[1]);
-        l2Reward.resumeUnlockingCountdown(positionsToBeModified);
+            scenario.lastClaimedAmount += scenario.balances[i] - scenario.oldBalances[i];
+        }
+    }
 
-        vm.warp(19740 days + 89 days);
-        // Pauses stake 5 on day 89
-        positionsToBeModified[0] = lockIDs[4];
-        vm.prank(stakers[4]);
-        l2Reward.pauseUnlocking(positionsToBeModified);
+    function cacheBalances(Scenario memory scenario) private view {
+        for (uint8 i = 0; i < scenario.stakers.length; i++) {
+            scenario.oldBalances[i] = l2LiskToken.balanceOf(scenario.stakers[i]);
+        }
+    }
 
-        vm.warp(19740 days + 95 days);
-        // stake 7: (200 LSK, 200 days)
-        lockIDs[6] = when_stakerCreatesPosition(
-            stakers[6], Position({ amount: convertLiskToSmallestDenomination(200), duration: 200 })
-        );
+    function checkRewardsContractBalance(uint256 funds) public {
+        uint256 sumOfDailyRewards;
+        for (uint256 i = deploymentDate; i < l2Reward.todayDay(); i++) {
+            sumOfDailyRewards += l2Reward.dailyRewards(i);
+        }
 
-        checkConsistencyPendingUnlockDailyUnlocked(lockIDs, getLargestExpiryDate(lockIDs));
+        assertTrue(l2LiskToken.balanceOf(address(l2Reward)) > funds - sumOfDailyRewards);
+        assertEq(l2LiskToken.balanceOf(address(l2Reward)) / 10 ** 4, (funds - sumOfDailyRewards) / 10 ** 4);
+    }
 
-        vm.warp(19740 days + 100 days);
-        // Increases amount for stake 6
-        L2Reward.IncreasedAmount[] memory increasingAmounts = new L2Reward.IncreasedAmount[](1);
-        increasingAmounts[0].lockID = lockIDs[5];
-        increasingAmounts[0].amountIncrease = convertLiskToSmallestDenomination(50);
-        vm.startPrank(stakers[5]);
-        l2LiskToken.approve(address(l2Reward), increasingAmounts[0].amountIncrease);
-        l2Reward.increaseLockingAmount(increasingAmounts);
-        vm.stopPrank();
+    function verifyRewardsSurplusForLockedAmountBetweenDays(
+        uint256 dailyRewards,
+        uint256 totalAmountLocked,
+        uint256 startDay,
+        uint256 endDay,
+        Scenario memory scenario
+    )
+        public
+    {
+        uint256 rewardsCap = totalAmountLocked / 365;
+        uint256 expectedSurplus;
+        uint256 lastDay = deploymentDate + endDay;
 
-        checkConsistencyPendingUnlockDailyUnlocked(lockIDs, getLargestExpiryDate(lockIDs));
+        for (uint256 d = deploymentDate + startDay; d <= lastDay; d++) {
+            expectedSurplus += dailyRewards - rewardsCap;
```

Comment:
I understand now. You assume that there are no actions between `startDay` and `lastDay`, this is why this test works. Please add a comment here about this, so that it's easier to revisit this.

Reply:
Can this be marked as resolved?

Reply:
Have added the documentation, so resolving it.